### PR TITLE
Styling React Applications: Update styling recommendations

### PR DESCRIPTION
**Pull request by Siriuszx**
## Because

The lesson tells people that Tailwind CSS is a valid Vanilla CSS option which isn't completely untrue, however the general consensus is that students should refrain from utilizing CSS frameworks or component libraries and try writing their *own* css.

## This PR

- reworded lesson tip
- removed Tailwind CSS docs link from Additional Resources
- added navigation links that point to the list of styling options available for students to choose from

## Issue

Closes [#27867](https://github.com/TheOdinProject/curriculum/issues/27867)

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: react/the_react_ecosystem/styling_react_applications.md
Comment (MaoShizhong) [inline react/the_react_ecosystem/styling_react_applications.md:33]:
```suggestion
For learning purposes throughout this course, we recommend that you avoid using CSS frameworks or component libraries, and instead implement your component's styling from scratch i.e. use [CSS Modules](#vanilla-css) or a [CSS-in-JS](#css-in-js) option. 
```

Nit: "For learning purposes" (omitting "the") is more appropriate here.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Thanks for doing this PR. Would you mind fixing up the lint issues as well? They should be pretty self explanatory. Let me know if you run into any issues with this.

**Comment by Siriuszx** [comment]
Sorry for the force push, I thought it would overwrite my last commit and the message with typo if I use `git commit --amend`

**Review by MaoShizhong — COMMENTED** [review]
In all honesty, the more I think about it, the more unsure I am about having Tailwind within `Vanilla CSS`, because it isn't vanilla CSS by any means. CSS modules can be argued as "for all intents and purposes, vanilla CSS" since it's almost the same, just with very tiny differences like local scoping. 

I feel Tailwind ought to go in its own `### Utility frameworks` between `### CSS in JS` and `### Component libraries`.
- The Vanilla CSS section would need a very slight rewording as it's no long highlighting "a couple of things", and it won't need an ordered list anymore. CSS modules can just be a normal sentence/paragraph in it.
- The Tailwind list item can be made its own paragraph in its own `### Utility frameworks` section.

What do you think, @Siriuszx?

**Comment by Siriuszx** [comment]
I think the definition of "Vanilla CSS" is too vague and **IS** purely subjective in that context.

To avoid confusion in the future, I think it would be better if we just get rid of the term "Vanilla CSS" altogether and split the lesson into:
- CSS Modules
- CSS Utility Frameworks
- CSS-in-JS
- Component Libraries

**Comment by Siriuszx** [comment]
If we are on the same page about this could you please merge this PR into main and I will open a separate issue addressing this lesson's structure? It would be amazing if I get the chance to work on the next PR as well 🙂

**Comment by MaoShizhong** [comment]
@Siriuszx I think this is pretty directly related to the original issue and PR purpose, so I think it can be done in this PR instead of a separate one.

I like your idea about the sections so let's go with that.
I think the Introduction section could do with a slight update to include something about vanilla CSS potentially getting harder to manage in a larger project due to every file sharing a global scope. Therefore there are some popular alternatives in common use.

I think that'll help set the scene for why something like css modules is so useful because of the local scoping, while being otherwise almost vanilla css.

I'll leave that with you, and we can always refine in review if we need to.

**Comment by Siriuszx** [comment]
@MaoShizhong sorry if this is taking too long, it says that you've "self-assigned this". Should I still work on this issue or not?

**Comment by MaoShizhong** [comment]
Oh yeah, carry on with it @Siriuszx. The PR assignment is a maintainer thing, you can ignore it

File: react/the_react_ecosystem/styling_react_applications.md
Comment (MaoShizhong) [inline react/the_react_ecosystem/styling_react_applications.md:13]:
```suggestion
Regular CSS is the simplest way to style. CSS Modules let you write CSS style declarations that are scoped locally, which means that we finally no longer have to worry about our class names potentially conflicting with other classes in the global scope. For example, we can have several CSS module files containing a `.wrapper` class, but each module's `.wrapper` will be treated as different to each other.
```
The above is purely a suggestion, so you don't have to take it at all. Just mainly feeling like it'd be best to just briefly explain how a class in one CSS module is different to the same class name in another CSS module.
I'm not sure the current example is 100% applicable (I'm being very nitpicky here!), since you may still want to use class names like `list-wrapper` and `item-wrapper` between/within CSS modules for various reasons.

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Love this! Just a few nits/suggestions below.

File: react/the_react_ecosystem/styling_react_applications.md
Comment (Siriuszx) [reply react/the_react_ecosystem/styling_react_applications.md:13]: You are right, I guess it could be even redundant. People will find explanations for what it really is when they eventually get to the assignment.

Comment (MaoShizhong) [reply react/the_react_ecosystem/styling_react_applications.md]: Agreed, especially when the documentation for CSS modules is already linked as an assigned reading.
We could probably just leave it at something like
> Regular CSS is the simplest way to style. CSS Modules let you write locally scoped CSS style declarations, preventing many issues with clashing class names in the global scope.

Comment (Siriuszx) [reply react/the_react_ecosystem/styling_react_applications.md]: Already done! :)

> Agreed, especially when the documentation for CSS modules is already linked as an assigned reading. We could probably just leave it at something like
> 
> > Regular CSS is the simplest way to style. CSS Modules let you write locally scoped CSS style declarations, preventing many issues with clashing class names in the global scope.

**Comment by Siriuszx** [comment]
It seems there is something wrong with my pr. Please don't merge @MaoShizhong, I will fix it.

**Review by MaoShizhong — APPROVED** [review]
:rocket:

**Pull request merged**
